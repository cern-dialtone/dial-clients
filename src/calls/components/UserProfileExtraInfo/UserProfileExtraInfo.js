import React from 'react';
import { Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';

export default function UserProfileExtraInfo({
  mail,
  physicalDeliveryOfficeName
}) {
  return (
    <ul>
      <li>
        <Icon name="mail" /> {mail}
      </li>
      <li>
        <Icon name="pin" /> {physicalDeliveryOfficeName}
      </li>
    </ul>
  );
}

UserProfileExtraInfo.propTypes = {
  mail: PropTypes.string.isRequired,
  physicalDeliveryOfficeName: PropTypes.string.isRequired
};
