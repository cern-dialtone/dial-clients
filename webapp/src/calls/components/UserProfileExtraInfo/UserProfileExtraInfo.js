import React from "react";
import { Icon } from "semantic-ui-react";
import PropTypes from "prop-types";

export function UserProfileExtraInfo(props) {
  return (
    <ul>
      <li>
        <Icon name={"mail"} /> {props.mail}
      </li>
      <li>
        <Icon name={"pin"} /> {props.physicalDeliveryOfficeName}
      </li>
    </ul>
  );
}

UserProfileExtraInfo.propTypes = {
  mail: PropTypes.any,
  physicalDeliveryOfficeName: PropTypes.any
};
