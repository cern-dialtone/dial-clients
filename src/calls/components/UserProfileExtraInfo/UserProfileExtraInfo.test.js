import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import UserProfileExtraInfo from './UserProfileExtraInfo';

global.remote = {
  require: jest.fn()
};

describe('UserProfileExtraInfo', () => {
  it('renders the component', () => {
    const { container } = render(
      <UserProfileExtraInfo
        mail="johndoe1@cern.ch"
        physicalDeliveryOfficeName="28/1-002"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders component and includes text', async () => {
    const { getByText } = render(
      <UserProfileExtraInfo
        mail="johndoe@cern.ch"
        physicalDeliveryOfficeName="28/1-002"
      />
    );

    await waitForElement(() => getByText(/johndoe@cern.ch/i));
    await waitForElement(() => getByText('28/1-002'));
  });
});
