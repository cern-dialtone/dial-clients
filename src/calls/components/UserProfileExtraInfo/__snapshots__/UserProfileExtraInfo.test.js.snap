// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileExtraInfo renders the component 1`] = `
<ul>
  <li>
    <i
      aria-hidden="true"
      class="mail icon"
    />
     
    johndoe1@cern.ch
  </li>
  <li>
    <i
      aria-hidden="true"
      class="pin icon"
    />
     
    28/1-002
  </li>
</ul>
`;
